### PR TITLE
Test and fix multiple precision in matrix_op

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this pull request

| Package Name | master | stable |
|--|--|--|
| AmplNLReader | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002300) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002301) | 
| CUTEst | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002302) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002303) | 
| CaNNOLeS | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002304) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002305) | 
| NLPModelsIpopt | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002306) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002307) | 
| NLPModelsJuMP | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002308) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002309) | 
| QuadraticModels | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002310) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002311) | 
| SolverTools | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002312) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655002313) |

File: src/NLPModels.jl [hunk 1]
```diff
@@ -300,7 +300,7 @@ The resulting object may be used as if it were a matrix, e.g., `J * v` or
 function jac_op(nlp :: AbstractNLPModel, x :: AbstractVector)
   prod = @closure v -> jprod(nlp, x, v)
   ctprod = @closure v -> jtprod(nlp, x, v)
-  return LinearOperator{Float64}(nlp.meta.ncon, nlp.meta.nvar,
+  return LinearOperator{eltype(x)}(nlp.meta.ncon, nlp.meta.nvar,
                                  false, false, prod, ctprod, ctprod)
```
Comment (dpo) [inline src/NLPModels.jl]: fix indentation here

File: src/NLPModels.jl [hunk 2]
```diff
@@ -316,7 +316,7 @@ function jac_op!(nlp :: AbstractNLPModel, x :: AbstractVector,
                  Jv :: AbstractVector, Jtv :: AbstractVector)
   prod = @closure v -> jprod!(nlp, x, v, Jv)
   ctprod = @closure v -> jtprod!(nlp, x, v, Jtv)
-  return LinearOperator{Float64}(nlp.meta.ncon, nlp.meta.nvar,
+  return LinearOperator{eltype(x)}(nlp.meta.ncon, nlp.meta.nvar,
                                  false, false, prod, ctprod, ctprod)
```
Comment (dpo) [inline src/NLPModels.jl]: and here

File: src/NLPModels.jl [hunk 3]
```diff
@@ -489,7 +489,7 @@ $(OBJECTIVE_HESSIAN).
 """
 function hess_op(nlp :: AbstractNLPModel, x :: AbstractVector; obj_weight::Real=one(eltype(x)))
   prod = @closure v -> hprod(nlp, x, v; obj_weight=obj_weight)
-  return LinearOperator{Float64}(nlp.meta.nvar, nlp.meta.nvar,
+  return LinearOperator{eltype(x)}(nlp.meta.nvar, nlp.meta.nvar,
                                  true, true, prod, prod, prod)
```
Comment (dpo) [inline src/NLPModels.jl]: and here

File: src/NLPModels.jl [hunk 4]
```diff
@@ -503,7 +503,7 @@ $(LAGRANGIAN_HESSIAN).
 """
 function hess_op(nlp :: AbstractNLPModel, x :: AbstractVector, y :: AbstractVector; obj_weight::Real=one(eltype(x)))
   prod = @closure v -> hprod(nlp, x, y, v; obj_weight=obj_weight)
-  return LinearOperator{Float64}(nlp.meta.nvar, nlp.meta.nvar,
+  return LinearOperator{eltype(x)}(nlp.meta.nvar, nlp.meta.nvar,
                                  true, true, prod, prod, prod)
```
Comment (dpo) [inline src/NLPModels.jl]: etc.

**Comment by abelsiqueira** [comment]
Thanks. Fixed.

**Comment by abelsiqueira** [comment]
:robot: Testing breakage of this pull request

| Package Name | master | stable |
|--|--|--|
| AmplNLReader | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064009) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064010) | 
| CUTEst | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064011) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064012) | 
| CaNNOLeS | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064013) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064014) | 
| NLPModelsIpopt | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064015) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064016) | 
| NLPModelsJuMP | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064017) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064018) | 
| QuadraticModels | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064019) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064020) | 
| SolverTools | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064021) | [](https://travis-ci.org/JuliaSmoothOptimizers/NLPModels.jl/jobs/655064022) |

**Comment by dpo** [comment]
Thank you.

**Pull request merged**